src/sentry/api/serializers/models/organization_member.py

```diff
@@ -78,3 +78,39 @@ def serialize(self, obj, attrs, user):
         d['teams'] = attrs.get('teams', [])
 
         return d
+
+
+class OrganizationMemberWithProjectsSerializer(OrganizationMemberSerializer):
+    def __init__(self, *args, **kwargs):
+        self.project_ids = set(kwargs.pop('project_ids', []))
+        super(OrganizationMemberWithProjectsSerializer, self).__init__(*args, **kwargs)
+
+    def get_attrs(self, item_list, user):
+        attrs = super(OrganizationMemberWithProjectsSerializer, self).get_attrs(
+            item_list,
+            user,
+        )
+        # Note: For this to be efficient, call
+        # `.prefetch_related('teams', 'teams__projectteam_set')` on your
+        # queryset before using this serializer
+        for org_member in item_list:
+            projects = set()
+            for team in org_member.teams.all():
+                # Filter in python here so that we don't break the prefetch
+                if team.status != TeamStatus.VISIBLE:
+                    continue
+
+                for project_team in team.projectteam_set.all():
+                    if project_team.project_id in self.project_ids:
+                        projects.add(project_team.project_id)
```

Comment:
Do you want the id or the slug? Our project endpoints want the slug not the id, and the UI generally uses project.slug as `projectId`.

Reply:
Good point, I'll make this the slug.